internal/controller/common/riders/rider.go

```diff
@@ -172,3 +174,31 @@ func GetRidersFromK8S(ctx context.Context, namespace string, ridersList []apiv1.
 	}
 	return existingRiders, nil
 }
+
+func VerifyRiders(riders []Rider) error {
+
+	// get global configmap
+	globalConfig, err := config.GetConfigManagerInstance().GetConfig()
+	if err != nil {
+		return err
+	}
+
+	// parse permitted Riders and create resource filter
+	riderRules, err := sync.ParseResourceFilter(globalConfig.PermittedRiders)
```

Comment:
not related to your code, but I noticed this function returns a pointer - I wonder if we can change that to just return an array slice instead? then we don't need to check for nil over here

Reply:
Was able to make this change to have this function no longer return a pointer.